scc/light_client/light_client_state/light_client_state.go

```diff
@@ -0,0 +1,151 @@
+package lc_state
+
+import (
+	"fmt"
+
+	"github.com/0xsoniclabs/consensus/inter/idx"
+	"github.com/0xsoniclabs/sonic/scc"
+	"github.com/0xsoniclabs/sonic/scc/cert"
+	"github.com/0xsoniclabs/sonic/scc/light_client/provider"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// State holds the current state of the light client.
+type State struct {
+	committee  scc.Committee
+	period     scc.Period
+	headNumber idx.Block
+	headHash   common.Hash
+	headRoot   common.Hash
+	hasSynced  bool
+}
+
+// NewState creates a new State with the given committee.
+// The given committee must be a valid committee and it is expected to be the
+// committee for the genesis period.
+func NewState(committee scc.Committee) *State {
+	return &State{
+		committee: committee,
+	}
+}
+
+// Head returns the block number of the latest known block.
+func (s *State) Head() (idx.Block, bool) {
+	return s.headNumber, s.hasSynced
+}
+
+// StateRoot returns the state root of the latest known block.
+func (s *State) StateRoot() (common.Hash, bool) {
```

Comment:
nit: document that the second return value indicates whether the provided root hash can be trusted.

Reply:
https://github.com/0xsoniclabs/sonic/pull/100/commits/504dff3b0ec1c192f603314d0fd8d1fb1b537d48